models/_sector/dex/trades/avalanche_c/platforms/pharaoh_avalanche_c_base_trades.sql

```diff
@@ -0,0 +1,66 @@
+{{ config(
+    schema = 'pharaoh_avalanche_c'
+    , alias = 'base_trades'
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , incremental_strategy = 'merge'
+    , unique_key = ['tx_hash', 'evt_index']
+    , incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+    )
+}}
+
+WITH dexs AS
+(
+    SELECT
+        t.evt_block_time AS block_time
+        ,t.recipient AS taker
+        ,CAST(NULL AS VARBINARY) AS maker
+        ,CASE
+            WHEN router.evt_tx_hash IS NULL
+                THEN CASE WHEN amount0 < INT256 '0' THEN abs(amount0) ELSE abs(amount1) END
+                ELSE router.amountOut
+            END AS token_bought_amount_raw
+        ,CASE
+            WHEN router.evt_tx_hash IS NULL
+                THEN CASE WHEN amount0 < INT256 '0' THEN abs(amount1) ELSE abs(amount0) END
+                ELSE router.inputAmount
+            END AS token_sold_amount_raw
+        ,NULL AS amount_usd
+        ,CASE WHEN amount0 < INT256 '0' THEN f.token0 ELSE f.token1 END AS token_bought_address
+        ,CASE WHEN amount0 < INT256 '0' THEN f.token1 ELSE f.token0 END AS token_sold_address
```

Comment:
If the latest commit does not fix the known issue, then we may fix these two lines, and revisit the source table used.

Reply:
Hi @discochuck, let's have a look on this trade:
https://snowtrace.io/tx/0xbf47a27f6b8bd7c2ae40b6d97a2307d8285f40a8c38644116c0037bcab54d27f?chainId=43114

It would be token `COQ` in exchange for `KIMBO` is I am not wrong.
The `pharaoh_avalanche_c.ClPool_evt_Swap` recorded it as `COQ <-> WAVAX` trade.

Reply:
yep, coq->kimbo

hrmmm, that's very weird. The project pharaoh is forked from (Ramses' on arbitrum) hasn't had any of these issues in the context of their dune spells. 

the only difference is that when we were trying to add all the contracts almost none of them were verified due to insane gas and snowtrace having just be migrated fro etherscan to whatever it is now. 

so I had to manually apply a number of ABIs. tho it seems like we'd be seeing way more issues than that if that were the cause.

Reply:
I'll take a look at this in the am before work.

Reply:
I've asked @echovl to take a look at this PR. He's from the ramses team and understands the inner workings of the tech stack way better than I. This is way outside my expertise.

Reply:
> Hi @discochuck, let's have a look on this trade: https://snowtrace.io/tx/0xbf47a27f6b8bd7c2ae40b6d97a2307d8285f40a8c38644116c0037bcab54d27f?chainId=43114
> 
> It would be token `COQ` in exchange for `KIMBO` is I am not wrong. The `pharaoh_avalanche_c.ClPool_evt_Swap` recorded it as `COQ <-> WAVAX` trade.


So, based on that inconsistency where do you think the problem is? 

I started digging into the Ramses PR for their dex.trades_beta inclusion, and they don't seem to need to left join in on OdosRouterV2_evt_Swap (or the equiv over on Arbitrum) 

Odos isn't included in the aggregator spell as the odos_trades had some sort of issue and was only on avalanche if I recall. I'm wondering if the inclusion of that is somehow scuffing the data?

Here's how Ramses did it over on Arb: https://github.com/echovl/spellbook/blob/1b24a7eec9a95a085ce6642dc9cb7e219203384f/models/_sector/dex/trades/arbitrum/platforms/ramses_arbitrum_base_trades.sql

I think it had something to do with certain trades not being found in the base ClPool_evt_Swap, but I can't recall the specifics.

Reply:
I may use the information in `glacier_avalanche_c.OdosRouterV2_evt_Swap` to fix this line. But still not sure about the final output will be correct.